include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -270,6 +270,11 @@ ORT_RUNTIME_CLASS(CUDAProviderOptionsV2);
 ORT_RUNTIME_CLASS(Op);
 ORT_RUNTIME_CLASS(OpAttr);
 
+#ifdef ENABLE_TRAINING_ON_DEVICE
```

Comment:
same, about conditioning on enable_training as well. Either that or we set enable_training whenever enable_training_on_device is set- which might be better considering we will refactor the build to be lightweight later on and will be separate from enable_training

Reply:
Specific to this file - we want to include these apis in the build only when --enable_training_on_device is enabled. These apis are not required when --enable_training flag is used.